evm/src/main/java/org/hyperledger/besu/evm/processor/ContractCreationProcessor.java

```diff
@@ -101,7 +102,17 @@ public void start(final MessageFrame frame, final OperationTracer operationTrace
       final MutableAccount contract = frame.getWorldUpdater().getOrCreate(contractAddress);
       long statelessGasCost =
           evm.getGasCalculator().proofOfAbsenceCost(frame, contract.getAddress());
+      if (handleInsufficientGas(
+          frame,
+          statelessGasCost,
+          () ->
+              String.format(
+                  "Not enough gas to cover proof of absence fee for %s: remaining gas = %d < %d = creation fee",
+                  frame.getContractAddress(), frame.getRemainingGas(), statelessGasCost))) {
+        return;
```

Comment:
do we need to return or set the state of the frame as we do here ?
frame.setExceptionalHaltReason(Optional.of(ExceptionalHaltReason.ILLEGAL_STATE_CHANGE));
frame.setState(MessageFrame.State.EXCEPTIONAL_HALT);

Reply:
is it Illegal state change? I think it is insufficient gas. It's already done inside `handleInsufficientGas`